src/Functional/functions.php

```diff
@@ -589,6 +590,8 @@ function head($list)
     foreach ($list as $item) {
         return $item;
     }
+
+    return null; // this can happen when the list is empty
```

Comment:
IMHO comment is not necessary.
But making explicit `return null` is good thing.